File: packages/snowplow/android/src/main/java/com/suamusica/snowplow/SnowplowTrackerBuilder.java

package com.suamusica.snowplow;

import com.snowplowanalytics.snowplow.configuration.*;
import com.snowplowanalytics.snowplow.controller.*;
import com.snowplowanalytics.snowplow.*;
import com.snowplowanalytics.snowplow.network.HttpMethod;
import com.snowplowanalytics.snowplow.tracker.LogLevel;
import com.snowplowanalytics.snowplow.util.TimeMeasure;

import android.content.Context;

import java.util.concurrent.TimeUnit;

public class SnowplowTrackerBuilder {
    public static final String HTTPS_SNOWPLOW_COLLECTOR_URL_COM = "https://snowplow.suamusica.com.br";

    public static TrackerController getTracker(Context context) {
        NetworkConfiguration networkConfig = new NetworkConfiguration(HTTPS_SNOWPLOW_COLLECTOR_URL_COM, HttpMethod.POST);
        TrackerConfiguration trackerConfig = new TrackerConfiguration("1")
                .base64encoding(true)
                .sessionContext(true)
                .platformContext(true)
                .lifecycleAutotracking(true)
                .screenViewAutotracking(true)
                .screenContext(true)
                .applicationContext(true)
                .exceptionAutotracking(true)
                .installAutotracking(true);

        SessionConfiguration sessionConfig = new SessionConfiguration(
                new TimeMeasure(30, TimeUnit.SECONDS),
                new TimeMeasure(30, TimeUnit.SECONDS)
        );

        SubjectConfiguration subjectConfiguration = new SubjectConfiguration().userId("0");

        return Snowplow.createTracker(context,
                "sm",
                networkConfig,
                trackerConfig,
                sessionConfig,
                subjectConfiguration
        );

    }
}

File: packages/snowplow/ios/Classes/SnowplowTrackerBuilder.swift
import Foundation
import SnowplowTracker

class SnowplowTrackerBuilder: NSObject, RequestCallback {
    let kNamespace = "sm"
    let kAppId     = "1"
    
    func getTracker(_ url: String) -> TrackerController {
        var configurations : [Configuration] = []
        
        let networkConfig = NetworkConfiguration(endpoint: url, method: .post)
        
        configurations.append(networkConfig)
        
        let trackerConfiguration = TrackerConfiguration()
            .appId(kAppId)
            .base64Encoding(true)
            .sessionContext(true)
            .platformContext(true)
            .lifecycleAutotracking(true)
            .screenViewAutotracking(true)
            .screenContext(true)
            .applicationContext(true)
            .exceptionAutotracking(true)
            .installAutotracking(true)
        
        configurations.append(trackerConfiguration)
        
        let emitterConfiguration = EmitterConfiguration()
        emitterConfiguration.requestCallback = self
        
        configurations.append(emitterConfiguration)
    
        if #available(iOS 10, *) {
            let sessionConfiguration = SessionConfiguration(foregroundTimeout: Measurement(value: 30, unit: .seconds), backgroundTimeout: Measurement(value: 30, unit: .seconds))
            configurations.append(sessionConfiguration)
        } else {
            // Fallback on earlier versions
        }
        
        let subjectConfiguration = SubjectConfiguration()
        
        configurations.append(subjectConfiguration)
        
        return Snowplow.createTracker(namespace: kNamespace, network: networkConfig, configurations: configurations)
    }
        
    func onSuccess(withCount successCount: Int) {
        print("Success: \(successCount)")
    }
    
    func onFailure(withCount failureCount: Int, successCount: Int) {
        print("Failure: \(failureCount), Success: \(successCount)")
    }  
}
